File: go-service/models/stock_transfer.go
package models

import "time"

// StockTransfer represents a stock transfer between branches
type StockTransfer struct {
	ID                int        `db:"Id" json:"id"`
	TransferNumber    string     `db:"TransferNumber" json:"transferNumber"`
	FromBranchID      int        `db:"FromBranchId" json:"fromBranchId"`
	ToBranchID        int        `db:"ToBranchId" json:"toBranchId"`
	RequestedByUserID int        `db:"RequestedByUserId" json:"requestedByUserId"`
	ApprovedByUserID  *int       `db:"ApprovedByUserId" json:"approvedByUserId,omitempty"`
	Status            string     `db:"Status" json:"status"`
	RequestedDate     time.Time  `db:"RequestedDate" json:"requestedDate"`
	CompletedDate     *time.Time `db:"CompletedDate" json:"completedDate,omitempty"`
	Notes             *string    `db:"Notes" json:"notes,omitempty"`
	CreatedAt         time.Time  `db:"CreatedAt" json:"createdAt"`
	UpdatedAt         *time.Time `db:"UpdatedAt" json:"updatedAt,omitempty"`
}

// StockTransferItem represents an item in a stock transfer
type StockTransferItem struct {
	ID                  int       `db:"Id" json:"id"`
	StockTransferID     int       `db:"StockTransferId" json:"stockTransferId"`
	ProductID           int       `db:"ProductId" json:"productId"`
	QuantityRequested   int       `db:"QuantityRequested" json:"quantityRequested"`
	QuantityTransferred *int      `db:"QuantityTransferred" json:"quantityTransferred,omitempty"`
	CreatedAt           time.Time `db:"CreatedAt" json:"createdAt"`
}

File: frontend/src/app/core/domain/stock-transfer/stock-transfer.ts
import { Branch } from '../branch/branch';
import { User } from '../user/user';
import { StockTransferStatus } from '../../enums/stock-transfer-status';

/**
 * Represents a Stock Transfer Item
 */
export interface StockTransferItem {
  /** ID */
  id: number;

  /** Stock Transfer ID */
  stockTransferId: number;

  /** Product ID */
  productId: number;

  /** Quantity Requested */
  quantityRequested: number;

  /** Quantity Transferred */
  quantityTransferred?: number | null;

  /** Created At */
  createdAt?: Date;

  /** Updated At */
  updatedAt?: Date;
}

/**
 * Represents a Stock Transfer
 */
export interface StockTransfer {
  /** ID */
  id: number;

  /** Transfer Number - unique identifier */
  transferNumber: string;

  /** From Branch ID - source branch */
  fromBranchId: number;

  /** To Branch ID - destination branch */
  toBranchId: number;

  /** Requested By User ID - FK to User */
  requestedByUserId: number;

  /** Approved By User ID - FK to User */
  approvedByUserId?: number | null;

  /** Status */
  status: StockTransferStatus;

  /** Requested Date */
  requestedDate: Date;

  /** Completed Date */
  completedDate?: Date | null;

  /** Notes */
  notes?: string | null;

  /** From Branch */
  fromBranch?: Branch;

  /** To Branch */
  toBranch?: Branch;

  /** Requested By User */
  requestedByUser?: User;

  /** Approved By User */
  approvedByUser?: User | null;

  /** Items - transfer items */
  items?: StockTransferItem[];

  /** Created At */
  createdAt?: Date;

  /** Updated At */
  updatedAt?: Date;

  /** Deleted At */
  deletedAt?: Date | null;
}
